### PR TITLE
Managing the deletion of AI configurations

### PR DESCRIPTION
**Pull request by assouktim**

### CODE REVIEW

File: bot/admin/server/src/main/kotlin/BotAdminService.kt [hunk 1]
```diff
@@ -1131,6 +1135,21 @@ object BotAdminService {
         ).forEach { story ->
             storyDefinitionDAO.delete(story)
         }
+
+        // delete the RAG configuration
+        ragConfigurationDAO.findByNamespaceAndBotId(app.namespace, app.name)?.let {
+            ragConfigurationDAO.delete(it._id)
+        }
+
+        // delete the RAG configuration
```
Comment (morgandiverrez) [inline bot/admin/server/src/main/kotlin/BotAdminService.kt:1144]:
```suggestion
        // delete the Observability configuration
```

File: bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt [hunk 2]
```diff
@@ -37,11 +37,25 @@ object SentenceGenerationService {
...
      * Get the LLM Sentence Generation configuration
+     * @param namespace: the namespace
+     * @param botId: the bot ID
      */
     fun getSentenceGenerationConfiguration(namespace: String, botId: String): BotSentenceGenerationConfiguration? {
         return sentenceGenerationConfigurationDAO.findByNamespaceAndBotId(namespace, botId)
     }
 
+    /**
+     * Deleting the Observability Configuration
```
Comment (morgandiverrez) [inline bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt:48]:
```suggestion
     * Deleting the LLM Sentence Generation Configuration
```

File: bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt [hunk 3]
```diff
@@ -37,11 +37,25 @@ object SentenceGenerationService {
...
+     * @param namespace: the namespace
+     * @param botId: the bot ID
+     */
+    fun deleteConfig(namespace: String, botId: String) {
+        val sentenceGenerationConfiguration = sentenceGenerationConfigurationDAO.findByNamespaceAndBotId(namespace, botId)
+            ?: WebVerticle.badRequest("No Observability configuration is defined yet [namespace: $namespace, botId: $botId]")
```
Comment (morgandiverrez) [inline bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt:54]:
```suggestion
            ?: WebVerticle.badRequest("No LLM Sentence Generation configuration is defined yet [namespace: $namespace, botId: $botId]")
```

File: bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt [hunk 4]
```diff
@@ -37,11 +37,25 @@ object SentenceGenerationService {
...
+        logger.info { "Deleting the Observability Configuration [namespace: $namespace, botId: $botId]" }
```
Comment (morgandiverrez) [inline bot/admin/server/src/main/kotlin/service/SentenceGenerationService.kt:55]:
```suggestion
        logger.info { "Deleting the LLM Sentence Generation Configuration [namespace: $namespace, botId: $botId]" }
```

**Review by morgandiverrez — CHANGES_REQUESTED** [review]

**Review by morgandiverrez — APPROVED** [review]

**Review by Benvii — APPROVED** [review]
Ok for me do we need to already create a ticket for vector db settings deletion ? 

2 options : 
- merging this PR first, rebasing https://github.com/theopenconversationkit/tock/pull/1675 and adding the deletion of vector DB settings.
- merging vector DB settings first then rebasing this PR and adding the deletion settings

**Pull request merged**